packages/skygear-core/lib/database.js

```diff
@@ -295,20 +314,27 @@ export class Database {
    * The delete will be performed asynchronously and the returned promise will
    * be resolved when the operation completes.
    *
-   * @param  {Record|Record[]|QueryResult} _records - the records to delete
+   * @param  {Record|Record[]|QueryResult} attr - the records to delete
    * @return {Promise} promise
    */
-  async delete(_records) {
-    let records = _records;
-    let isQueryResult = records instanceof QueryResult;
-    if (!_.isArray(records) && !isQueryResult) {
-      records = [records];
-    }
-
-    let ids = _.map(records, perRecord => perRecord.id);
-    let payload = {
+  async delete(attr) {
+    const isQueryResult = attr instanceof QueryResult;
+    const isArrayLike = _.isArray(attr) || isQueryResult;
+    const records = isArrayLike ? attr : [attr];
+
+    const ids = _.map(
+      records,
+      perRecord => [perRecord.recordType, perRecord.recordID].join('/')
+    );
+    const recordIdentifiers = _.map(records, perRecord => ({
+      _id: [perRecord.recordType, perRecord.recordID].join('/'),
```

Comment:
for this one, you don’t need the backward-compatible `_id`, because the `records` field in the payload is entirely new.